### PR TITLE
allow sorting incident results by custom fields

### PR DESCRIPTION
**Pull request by diegocepedaw**
allows sorting incidents by specific fields. You can use the parameter `order_by` eg. `order_by=$FIELD` to sort the response by a specific field. Supported fields are:
        - `id`
        - `created`
        - `updated`
        - `owner`
        - `application`
        - `plan`
        - `context`
        - `active`
        
  Specify the order of the sort with the `order` parameter eg order=ASC. Supported values are:
        - ASC
        - DESC

### CODE REVIEW

File: src/iris/api.py [hunk 1]
```diff
@@ -678,7 +696,8 @@ def stream_incidents_with_context(results, title=False):
                 row['title'] = row['context'].get(title_variable_name)
             else:
                 row['title'] = None
-        yield row
+        results_list.append(row)
+    return results_list
```
Comment (jrgp) [inline src/iris/api.py:700]: With this change to this function, from being a generator yielding rows as it decodes them and instead accumulating them all at once in a list and return, it is no longer "streaming" the results and the name "stream_incidents_with_context" becomes misleading.

Can we either revert this change or rename the function?

File: src/iris/api.py [hunk 2]
```diff
@@ -1988,8 +2032,13 @@ def on_get(self, req, resp):
         if where:
             query = query + ' WHERE ' + ' AND '.join(where)
 
+        if order_by is not None:
+            query += ' ORDER BY %s %s' % (incident_order_by_fields[order_by], order)
+        else:
+            query += ' ORDER BY `incident`.`created` DESC, `incident`.`id` DESC'
```
Comment (jrgp) [inline src/iris/api.py:2038]: It looks like this always adds an ORDER BY clause even if one isn't requested. Depending on whether we have in index on the created field, this might be slower than expected. Would it make more sense to only order the result if it is explicitly requested?

File: src/iris/api.py (hunk 1 above)
Comment (diegocepedaw) [reply src/iris/api.py:700]: renamed

**Review by jrgp — APPROVED** [review]
